src/libcrun/utils.c

```diff
@@ -1916,35 +1930,57 @@ find_executable (const char *executable_path, const char *cwd)
     {
       ret = check_access (executable_path);
       if (ret == 0)
-        return xstrdup (executable_path);
-      return NULL;
+        {
+          *out = xstrdup (executable_path);
+          return 0;
+        }
```

Comment:
There seems to be no return here anymore which means even in the absolute path you fall back to looking up PATH. That seems wrong? should this set last_error = ret and then have en else branch for the PATH lookup?